packages/evm/contracts/payments/PaymentsBook.sol

```diff
@@ -0,0 +1,535 @@
+    */
+    constructor(Controller _controller, uint64 _periodDuration, uint16 _governorSharePct)
+        ControlledRecoverable(_controller)
+        public
+    {
+        // No need to explicitly call `Controlled` constructor since `ControlledRecoverable` is already doing it
+        require(_periodDuration > 0, ERROR_PERIOD_DURATION_ZERO);
+
+        periodDuration = _periodDuration;
+        _setGovernorSharePct(_governorSharePct);
+    }
+
+    /**
+    * @notice Pay `@tokenAmount(_token, _amount)` for `_payer` (`_data`)
+    * @param _token Address of the token being paid
+    * @param _amount Amount of tokens being paid
+    * @param _payer Address paying on behalf of
+    * @param _data Optional data
+    */
+    function pay(address _token, uint256 _amount, address _payer, bytes calldata _data) external payable {
+        (uint256 periodId, Period storage period) = _getCurrentPeriod();
+        require(_amount > 0, ERROR_PAYMENT_AMOUNT_ZERO);
+
+        // Update collected fees for the governor
+        uint256 governorFees = _amount.pct(governorSharePct);
+        period.governorFees[_token] = period.governorFees[_token].add(governorFees);
+
+        // Update collected fees for the jurors
+        uint256 jurorFees = _amount.sub(governorFees);
+        period.jurorFees[_token] = period.jurorFees[_token].add(jurorFees);
+
+        // Deposit tokens from sender to this contract
+        _deposit(msg.sender, _token, _amount);
+        emit PaymentReceived(periodId, _payer, _token, _amount, msg.sender, _data);
+    }
+
+    /**
+    * @notice Claim jurors fees for period #`_periodId` owed to `msg.sender`
+    * @param _periodId Identification number of the period which fees are claimed for
+    * @param _token Address of the token to be claimed
+    */
+    function claimJurorFees(uint256 _periodId, address _token) external {
+        require(_periodId < _getCurrentPeriodId(), ERROR_NON_PAST_PERIOD);
+
+        Period storage period = periods[_periodId];
+        require(!_hasClaimedJurorFees(period, msg.sender, _token), ERROR_JUROR_FEES_ALREADY_CLAIMED);
+
+        (uint64 periodBalanceCheckpoint, uint256 totalActiveBalance) = _ensurePeriodBalanceDetails(period, _periodId);
+        uint256 jurorActiveBalance = _getJurorActiveBalance(msg.sender, periodBalanceCheckpoint);
+        uint256 amount = _getJurorFees(period, _token, jurorActiveBalance, totalActiveBalance);
+        _claimJurorFees(period, _periodId, msg.sender, _token, amount);
+    }
+
+    /**
+    * @notice Claim juror fees for period #`_periodId` owed to `msg.sender`
+    * @dev It will ignore tokens that were already claimed without reverting
+    * @param _periodId Identification number of the period which fees are claimed for
+    * @param _tokens List of token addresses to be claimed
+    */
+    function claimManyJurorFees(uint256 _periodId, address[] calldata _tokens) external {
```

Comment:
I'm wondering if:

1. `claimGuardianShare()` and `transferGovernorShare()` are both redundant, since there are array versions available (and the gas overhead is probably only in the hundreds of gas, if even that)
2. Add the ability to claim by signature. This would let us build additional meta-delegation wrappers as necessary to bundle the txs for multiple periods.

Reply:
Addressed in 51b7ed3c9324e62bd2b9bc9b85047db0c062ac7b and e6fe18fc83ecf84deecb75da5e5f13ab3b76b411 respectively